src/pudl/transform/eia.py

```diff
@@ -312,45 +312,36 @@ def _compile_all_entity_records(
-            df = df[(base_cols + cols)]
-            df = df.dropna(subset=id_cols)
-            # add a column with the table name so we know its origin
-            df["table"] = table_name
-            dfs.append(df)
-            # check if there are any columns that should be renamed and harvested
+        if "annual" not in table_name:
+            if set(base_cols).issubset(transformed_df.columns):
+                logger.debug(f"        {table_name}...")
+                # create a copy of the df to muck with
+                df = transformed_df.copy()
+                # we know these columns must be in the dfs
+                cols = []
+                # check whether the columns are in the specific table
+                for column in static_cols + annual_cols:
+                    if column in df.columns:
+                        cols.append(column)
+                df = df[(base_cols + cols)]
+                df = df.dropna(subset=id_cols)
+                # add a column with the table name so we know its origin
+                df["table"] = table_name
+                dfs.append(df)
+            # check if there are columns that should be renamed and harvested
             # as an additional table
-            if map_cols_dict and (
-                set(map_cols_dict.keys()) & set(transformed_df.columns)
-            ):
-                mapped_df = transformed_df.copy()
-                map_cols_present_dict = {
-                    col: col_rename
-                    for col, col_rename in map_cols_dict.items()
-                    if col in mapped_df.columns
-                }
-                logger.debug(f"map_cols_present_dict: {map_cols_present_dict}")
-                # base cols that aren't included by a soon to be renamed column
-                base_cols_to_add = set(base_cols) - set(map_cols_present_dict.values())
-                mapped_df = mapped_df[
-                    list(base_cols_to_add) + list(map_cols_present_dict.keys())
-                ]
-                logger.debug(f"mapped_df columns: {mapped_df.columns}")
-                mapped_df = mapped_df.rename(columns=map_cols_present_dict)
-                mapped_df = mapped_df.dropna(subset=id_cols)
-                mapped_df["table"] = table_name + "_mapped"
-                dfs.append(mapped_df)
+            if map_cols_dict:
+                base_cols_to_add = set(base_cols) - set(map_cols_dict.values())
+                if base_cols_to_add.union(set(map_cols_dict.keys())).issubset(
+                    transformed_df.columns
+                ):
+                    mapped_df = transformed_df[
+                        list(base_cols_to_add) + list(map_cols_dict.keys())
+                    ]
+                    mapped_df = mapped_df.rename(columns=map_cols_dict)
+                    mapped_df = mapped_df.dropna(subset=id_cols)
+                    mapped_df["table"] = table_name + "_mapped"
+                    dfs.append(mapped_df)
```

Comment:
i think i still see this as adding two sets of columns when there are `owner` and `operator` columns in one table. or at least it does not seem like it is going to be able to pick up both id's from one table.

hm okay nvm so the ownership table now has `owner` and `operator` prefixes and the `owner` prefixed columns are the ones getting renamed and added. the ownership table skips over the entire `if set(base_cols).issubset(transformed_df.columns):` section bc `base_cols` includes `utility_id_eia` and that is actually not yet in the table.

Reply:
i think it would still generally be preferable if we could grab both sets of ids from this ownership table for the harvesting process. although if there are truly no new operator ids' from this table i guess it is fine to basically ignore the operator column from the ownership table.

Reply:
> the ownership table skips over the entire `if set(base_cols).issubset(transformed_df.columns):` section bc `base_cols` includes `utility_id_eia` and that is actually not yet in the table.

Yep exactly.



> i think it would still generally be preferable if we could grab both sets of ids from this ownership table for the harvesting process. although if there are truly no new operator ids' from this table i guess it is fine to basically ignore the operator column from the ownership table.

About to push a change that should do this.